Plugin/CppApi/include/AbstractTool.h

```diff
@@ -39,11 +40,14 @@ class TOOLKIT_EXPORT AbstractTool : public QObject
   virtual QString toolName() const = 0;
   virtual bool handleClick(const Point& pos);
 
+  virtual void setProperties(const QVariantMap& properties);
+
   virtual void setActive(bool active);
   bool isActive() const;
 
 signals:
   void errorOccurred(const Error& error);
+  void propertyUpdated(const QString& propertyName, const QVariant& propertyValue);
```

Comment:
@lsmallwood Can we call this `propertyChanged` instead?  Usually `*Changed` is what I typically see.